### ISSUE TITLE
Feature Request: Show suggestions when `npm run script` can't find `script`

### ISSUE THREAD

**Issue by AlvaroBernalG** [npm/npm#20858]
Hello good people of npm,

#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] npm is producing incorrect or undesirable behavior.
  - [x] Other (_see below for feature requests_):

#### What's the feature?

Whenever `npm run 'script'` is unable to find `'script'`, display a list with all possible available scripts.

Example: 

<img width="464" alt="screen shot 2018-05-31 at 23 38 06" src="https://user-images.githubusercontent.com/12004383/40812041-bd1e417c-652b-11e8-9853-0b4d89d1c303.png">

#### What problem is the feature intended to solve?

Improves the user experience.

#### Is this feature similar to an existing feature in another tool?

git and npm does a similar thing whenever a built in command is misspelled:

<img width="456" alt="screen shot 2018-05-31 at 21 57 06" src="https://user-images.githubusercontent.com/12004383/40807953-9c5df3dc-651d-11e8-844a-f6f5d522f762.png">

<img width="533" alt="screen shot 2018-05-31 at 22 52 53" src="https://user-images.githubusercontent.com/12004383/40810932-3afd8e2c-6527-11e8-979c-3143775011e2.png">

#### Is this a feature you're prepared to implement, with support from the npm CLI team?

Yes.

Thanks for all the awesome work.

**Comment by kenany** [comment]
https://github.com/npm/rfcs/pull/9?

**Comment by AlvaroBernalG** [comment]
@KenanY Yes, that is exactly what I am suggesting. Glad that the feature is already being created!.
